YahooFantasy.mjs

```diff
@@ -39,50 +55,225 @@ class YahooFantasy {
     // this.users = new Users(); // TODO
 
     this.yahooUserToken = null;
+    this.yahooRefreshToken = null;
+  }
+
+  // oauth2 authentication function -- follow redirect to yahoo login
+  auth(res) {
+    const authData = stringify({
+      client_id: this.CONSUMER_KEY,
+      redirect_uri: this.REDIRECT_URI,
+      response_type: "code",
+    });
+
+    const options = {
+      hostname: "api.login.yahoo.com",
+      port: 443,
+      path: `/oauth2/request_auth?${authData}`,
+      method: "GET",
+    };
+
+    const authRequest = https.request(options, (authResponse) => {
+      authResponse.on("data", (d) => {
+        process.stdout.write(d);
+      });
+
+      authResponse.on("end", () => {
+        if (302 === authResponse.statusCode) {
+          res.redirect(authResponse.headers.location);
+        } else {
+          throw new Error("authentication error");
+        }
+      });
+    });
+
+    authRequest.on("error", (e) => {
+      throw new Error(e);
```

Comment:
You're throwing the error object, but you're not actually doing anything with it/surfacing it to the user other than "authentication error"

Reply:
Finally looking at this.

This would be an actual error that comes from the Yahoo! servers, my "Authentication Error" really *SHOULDN'T* happen, but I put it there as a fail safe. If there's an actual error with the auth flow it will give me a better error and description here...